src/Uart/Uart.c

```diff
@@ -314,24 +316,26 @@ handleTxInterrupt(Uart *const uart)
 	}
 }
 
-void
-Uart_handleInterrupt(Uart *const uart)
+bool
+Uart_handleInterrupt(Uart *const uart, int * errCode)
 {
-	uint32_t status = uart->reg->sr;
+	bool retValue = true;
+	uint32_t status = uart->reg->sr & uart->reg->imr;
 	uart->reg->cr = UART_CR_RSTSTA_MASK;
 	if ((status & UART_SR_RXRDY_MASK) != 0u)
-		handleRxInterrupt(uart);
+		retValue = handleRxInterrupt(uart, errCode);
```

Comment:
I would prefer if error return was close to the point where it occurred... I assume that status checks are valuable, and handling tx interrupt would also be nice. What do you think about moving the RX interrupt handling to the end, where the error can be returned immediatelly?

Reply:
That's the reason why I am not returning immediately here as well. 
No problem, I will do it. 
Let's hope that txHandler won't return any error in the future, though.. We would have the same problem.